x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/explain/CipherSuiteNames.java

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.ssl.explain;
+
+import org.elasticsearch.common.collect.ImmutableOpenIntMap;
+
+import java.util.Locale;
+
+/**
+ * Utility class to convert Cipher Suite codes (which are represented in a TLS stream as a 2 byte value) into a standard cipher suite name.
+ * This class is generated from the <a href="https://www.iana.org/assignments/tls-parameters/tls-parameters-4.csv">TLS code assignments</a>
+ * at IANA.
+ */
+class CipherSuiteNames {
```

Comment:
this probably needs updated for TLSv1.3. Its also a shame that the JDK doesn't make this available outside of a sun package class: https://github.com/AdoptOpenJDK/openjdk-jdk11u/blob/master/src/java.base/share/classes/sun/security/ssl/CipherSuite.java

Reply:
I think this is up to date - I generated the list from the latest CSV on the IANA site, and it has Ciphers from RFC8446 (TLS1.3) such as `TLS_AES_256_GCM_SHA384`